addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -111,9 +111,15 @@ export class DiscussCoreCommon {
                         .filter(({ type }) => type === "discuss.channel/leave")
                         .map(({ payload }) => payload.id)
                 );
+                const deletedMessagesIds = notifications
+                    .filter(({ type }) => type === "mail.message/delete")
+                    .map(({ payload }) => payload.message_ids)
+                    .flat(1);
```

Comment:
Why do you need to flat this array?

Reply:
`message_ids` is an array so the map creates an array of arrays

Reply:
Doesn't work but I found out about `flatMap`:
```js
                const deletedMessagesIds = notifications
                    .filter(({ type }) => type === "mail.message/delete")
                    .flatMap(({ payload }) => payload.message_ids)
```